Fabric.Authorization.API/RemoteServices/Identity/Providers/IdentityServiceProvider.cs

```diff
@@ -25,25 +28,49 @@ public IdentityServiceProvider(IHttpClientFactory httpClientFactory, IAppConfigu
 
         public async Task<IEnumerable<UserSearchResponse>> Search(string clientId, IEnumerable<string> userIds)
         {
-            // get all users in groups tied to clientRoles
-            var httpClient =
-                await _httpClientFactory.Create(
-                    new Uri(_appConfiguration.IdentityServerConfidentialClientSettings.Authority),
-                    IdentityScopes.ReadScope);
+            var userIdList = userIds.ToList();
+            if (userIdList.Count == 0)
+            {
+                return new List<UserSearchResponse>();
+            }
+
+            // TODO: clean this up / move to config
+            var settings = _appConfiguration.IdentityServerConfidentialClientSettings;
+            var httpClient = await Create(
+                $"{settings.Authority}/connect/token",
+                "fabric-authorization-client",
+                settings.ClientSecret,
+                new Uri(settings.Authority),
+                IdentityScopes.ReadScope);
 
             var request = new UserSearchRequest
             {
                 ClientId = clientId,
-                UserIds = userIds
+                UserIds = userIdList
             };
 
-            var response = await httpClient.PostAsync("/users",
+            var response = await httpClient.PostAsync("/api/users",
                 new StringContent(JsonConvert.SerializeObject(request), Encoding.UTF8, "application/json"));
 
             return response.StatusCode != HttpStatusCode.OK
                 ? new List<UserSearchResponse>()
                 : JsonConvert.DeserializeObject<IEnumerable<UserSearchResponse>>(await response.Content
                     .ReadAsStringAsync());
         }
+
+        private async Task<HttpClient> Create(string tokenUrl, string clientId, string clientSecret, Uri uri, string requestScope)
+        {
+            var tokenClient = new TokenClient(tokenUrl, clientId, clientSecret);
+            var response = await tokenClient.RequestClientCredentialsAsync(requestScope).ConfigureAwait(false);
+            return CreateWithAccessToken(uri, response.AccessToken);
```

Comment:
Did the HttpClientFactory.CreateWithAccessToken method not work? Let me know, I can check it out and fix it up.

Reply:
@michaelvidal24 I can use it, but I still have to create the HttpClientFactory and TokenClient and manually. I've changed the code to use the factory since that does eliminate some repeated code,  but we still have some refactor work to do. The issue is the HttpClientFactory accepts a tokenUrl that is pulled from configuration at app startup, which at the moment is just "http://localhost:5001". We need to append "/connect/token" to that URL. As part of the refactor work to segregate config settings, we can probably fix all of this so that IdentityServiceProvider uses the injected HttpClientFactory. We are currently losing the correlationToken with how we're doing it, but that should fixed with the config refactor too.

Reply:
@michaelvidal24, the other issue is the clientId in appsettings.json is incorrect for the Fabric auth client.